woocommerce/woocommerce-android
# Adds FluxC as composite build

_This PR is a carbon copy of its WPAndroid counterpart https://github.com/wordpress-mobile/WordPress-Android/pull/13001 with very minor WCAndroid specific changes._

This PR adds FluxC as a possible composite build. This is meant to make local development easier with the following main benefits:
* Being able to make changes to the FluxC codebase from the same Android Studio instance that is running WCAndroid
* Making changes to FluxC repo and immediately being able to use it within WCAndroid
* Being able to use IDE goodies such as "Find usages"

This is accomplished by:
* Including the FluxC project by its relative path by adding it to `local-builds.gradle` file which will make it available for use within the same Android Studio instance

Please see the notes in `local-builds.gradle-example` for more information.

**To test:**
* Copy `local-builds.gradle-example` as `local-builds.gradle` and update `localFluxCPath` to your liking
* Use `Navigate Class` shortcut to open `WCOrderStore` which is in FluxC and add a new method such as a summation
* Use `Navigate Class` shortcut to open `OrderDetailPresenter` which is in WCAndroid and use that function
* Run the project and verify that it works 🎉 
* Comment out `localFluxCPath` in `local-builds.gradle` file and verify that the build fails for missing the new method from `WCOrderStore`
* Discard the changes for `OrderDetailPresenter`, build the app and verify that it works

_P.S: Unfortunately making changes to FluxC is _very_ slow to compile. We can improve the builds and this won't be as big of an issue, but this probably won't happen at least until the end of year._

**Update release notes:**
